crates/hdk/src/hdk.rs

```diff
@@ -0,0 +1,356 @@
+use crate::prelude::*;
+
+#[cfg(feature = "mock")]
+use mockall::*;
+
+pub const HDK_NOT_REGISTERED: &str = "HDK not registered";
+
+/// This is a cell so it can be set many times.
+/// Every test needs its own mock so each test needs to set it.
+use core::cell::RefCell;
+thread_local!(pub static HDK: RefCell<Box<dyn HdkT>> = RefCell::new(Box::new(ErrHdk)));
```

Comment:
I think we could do something like this here:

```rust
#[cfg(not(feature = "mock"))]
thread_local!(pub static HDK: RefCell<Box<dyn HdkT>> = RefCell::new(Box::new(HostHdk)));


#[cfg(feature = "mock")]
thread_local!(pub static HDK: RefCell<Box<dyn HdkT>> = RefCell::new(Box::new(ErrHdk)));
```

and then remove the `init_hdk` part in each hdk function.